### PR TITLE
Improve it case

### PR DESCRIPTION
**Pull request by Sammers21**
Complete the `compare with helm generation results` puzzle. Also related to #44.

COULD BE MERGED ONLY AFTER #51. THE MERGE WILL BE BLOCKED UNTIL THEN

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#47](https://github.com/artipie/helm-adapter/issues/47) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/47e2ffad-f468-487d-b828-cced2a849a85); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by Vatavuk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/helm-adapter/issues/47#issuecomment-647417509))

@Vatavuk Job refused in 3 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/helm-adapter/issues/47#issuecomment-647417509))

@Vatavuk The user @vatavuk[/z](https://www.0crat.com/u/vatavuk) resigned from [#47](https://github.com/artipie/helm-adapter/issues/47), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#47](https://github.com/artipie/helm-adapter/issues/47) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/a87d0e99-6f61-41e6-94c0-77cdfe629edd); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by carlosmiranda** [comment]
@Sammers21 let me know when this is ready for merge

**Comment by Sammers21** [comment]
@carlosmiranda, Sure, I will

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@Sammers21 ping

**Comment by Sammers21** [comment]
@carlosmiranda, The PR is ready for review.

**Comment by Sammers21** [comment]
@g4s8, review the PR please

File: src/test/java/com/artipie/helm/SubmitChartITCase.java [hunk 1]
```diff
@@ -80,16 +101,51 @@ public void indexYamlIsCorrect() throws IOException, URISyntaxException {
...
+                new BlockingStorage(fls).value(new Key.From("index.yaml"))
+            );
+            LoggerFactory.getLogger(SubmitChartITCase.class).info(
+                "Generated by adapter index.yaml:\n{}",
+                adapter
+            );
+            final HelmContainer helm = new HelmContainer()
+                .withCommand("repo", "index", ".")
+                .withWorkingDirectory("/home/")
+                .withFileSystemBind("./src/test/resources", "/home");
```
Comment (g4s8) [inline src/test/java/com/artipie/helm/SubmitChartITCase.java:114]: @Sammers21 I don't think it's a good idea to use resources folder as a workdir for integration tests. Let's use `@TempDir` instead.

File: src/test/java/com/artipie/helm/SubmitChartITCase.java [hunk 2]
```diff
@@ -80,16 +101,51 @@ public void indexYamlIsCorrect() throws IOException, URISyntaxException {
...
             server.close();
             vertx.close();
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private static Map<String, Object> tomcatZeroEntry(final String yaml) {
+        return (
+            (ArrayList<Map<String, Object>>)
+                ((Map<String, Object>) new Yaml().<Map<String, Object>>load(yaml)
```
Comment (g4s8) [inline src/test/java/com/artipie/helm/SubmitChartITCase.java:144]: @Sammers21 can we somehow simplify this over-casting? It's really hard to read.

Comment (Sammers21) [reply src/test/java/com/artipie/helm/SubmitChartITCase.java:144]: I know it's hard, that why I decided to move this ugly code into a separated method. Unfortunately, there's no different way

Comment (g4s8) [reply src/test/java/com/artipie/helm/SubmitChartITCase.java:144]: @Sammers21 ok, so can you please put the documentation for this method?

File: src/test/java/com/artipie/helm/SubmitChartITCase.java (hunk 1 above)
Comment (Sammers21) [reply src/test/java/com/artipie/helm/SubmitChartITCase.java:114]: @g4s8, agree. I can create a puzzle for that

**Comment by Sammers21** [comment]
@g4s8, can we merge?

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (8 days), architects (@g4s8, @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by carlosmiranda** [comment]
@sereshqua this is stuck in QA queue.

**Comment by g4s8** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Pull request merged**